sycl/include/sycl/ext/intel/experimental/host_pipes.hpp

```diff
@@ -0,0 +1,93 @@
+//==---------------- pipes.hpp - SYCL pipes ------------*- C++ -*-----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// ===--------------------------------------------------------------------=== //
+
+#pragma once
+
+#include <sycl/context.hpp>
+#include <sycl/device.hpp>
+#include <sycl/ext/intel/experimental/pipe_properties.hpp>
+#include <sycl/ext/oneapi/properties/properties.hpp>
+#include <sycl/queue.hpp>
+#include <type_traits>
+
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+#include <xpti/xpti_data_types.h>
+#include <xpti/xpti_trace_framework.hpp>
+#endif
+
+namespace sycl {
+__SYCL_INLINE_VER_NAMESPACE(_V1) {
+namespace ext {
+namespace intel {
+namespace experimental {
+
+using default_pipe_properties =
+    decltype(sycl::ext::oneapi::experimental::properties(min_capacity<0>));
+
+template <class _name, class _dataT, class _propertiesT>
+class
+#ifdef __SYCL_DEVICE_ONLY__
+    [[__sycl_detail__::add_ir_attributes_global_variable("sycl-host-access",
+                                                         "readwrite")]]
+#endif
+    // TODO: change name to pipe, and merge into the existing pipe
+    // implementation
+    host_pipe {
+
+  struct
+// Commented out since the host_pipe attribute is not introduced by the front
+// end yet. Confirm with Rob
+#ifdef __SYCL_DEVICE_ONLY__
+      [[__sycl_detail__::add_ir_attributes_global_variable(
+          "sycl-host-pipe",
+          nullptr)]] [[__sycl_detail__::
+                           host_pipe]] [[__sycl_detail__::
+                                             global_variable_allowed]] // may
```

Comment:
To my knowledge it should be possible to do multiple attributes in the same `[[...]]`, which may make the formatting here a little more readable.